### ISSUE TITLE
How to notify BrowserSync of changes at end of gulp task?

### ISSUE THREAD

**Issue by chauncey-garrett** [BrowserSync/browser-sync#649]
Either I'm forgetting something or this is a feature request, but I'd like to be able to notify BrowserSync at the end of a gulp task. Can it be done? From the documentation, it seems I can only reload the browser from within the stream...

Here's a relevant portion of my gulp file:

```js
var browserSync = require('browser-sync').create(),
	$           = require('gulp-load-plugins')(),
	notify      = browserSync.notify;

gulp.task('json', function() {
	notify('JSON Optimizing'); // works fine

	return gulp.src(jsonSources)
		.pipe($.changed(jsonDest))
		.pipe($.if(env === 'production', $.jsonminify()))
		.pipe($.if(env === 'production', gulp.dest(jsonDest)))
		.pipe(browserSync.stream()) // reloads the browser
		.pipe($.size({title: 'json'}));
		// would like to notify via browser sync here—once task is complete
});
```

**Comment by chauncey-garrett** [comment]
I should add that I'd essentially like to be able to use BrowserSync in the same manner as [gulp-notify](https://www.npmjs.com/package/gulp-notify):

```js
var notify = require("gulp-notify");
gulp.src("./src/test.ext")
  .pipe(notify("Hello Gulp!"));
```

**Comment by shakyShane** [comment]
```js
var browserSync = require('browser-sync').create(),
    $           = require('gulp-load-plugins')(),
    notify      = browserSync.notify;

gulp.task('json', function() {
    notify('JSON Optimizing'); // works fine

    return gulp.src(jsonSources)
        .pipe($.changed(jsonDest))
        .pipe($.if(env === 'production', $.jsonminify()))
        .pipe($.if(env === 'production', gulp.dest(jsonDest)))
        .pipe(browserSync.stream()) // reloads the browser
        .pipe($.size({title: 'json'}))
        .on('end', function () {
            browserSync.notify('Task completed!')
         });
});
```

??

**Comment by shakyShane** [comment]
although, because you're reloading in the same stream, you may miss the notify message in the browser...

**Comment by chauncey-garrett** [comment]
@shakyShane That looks like it would work but I think you're right about the notify message being lost. Perhaps it would be an improvement  `.stream` if we had the option to include a message on reload?

**Comment by shakyShane** [comment]
I think that would be out of scope - what about doing the notification first, then reloading...

```js
var browserSync = require('browser-sync').create(),
    $           = require('gulp-load-plugins')(),
    notify      = browserSync.notify;

gulp.task('json', function() {
    notify('JSON Optimizing'); // works fine

    return gulp.src(jsonSources)
        .pipe($.changed(jsonDest))
        .pipe($.if(env === 'production', $.jsonminify()))
        .pipe($.if(env === 'production', gulp.dest(jsonDest)))
        .pipe($.size({title: 'json'}))
        .on('end', function () {
            browserSync.notify('Task completed!');
            setTimeout(function(){
                browserSync.reload();
            }, 2000);
         });
});
```

**Comment by chauncey-garrett** [comment]
Thanks for the code tip and your help. This works great!

**Comment by shakyShane** [comment]
@chauncey-garrett happy to help :)

Perhaps you could return the favour by [voting for us](https://thenetawards.com/vote/open-source/browsersync/) :p

**Comment by chauncey-garrett** [comment]
Just did! Thank you for such an excellent contribution. I use BrowserSync whenever I'm working on a site. It really makes a difference!

**Comment by shakyShane** [comment]
Great to hear :)
